src/dag/dag_block_manager.cpp

```diff
@@ -245,12 +245,14 @@ DagBlock DagBlockManager::popVerifiedBlock(bool level_limit, uint64_t level) {
       cond_for_verified_qu_.wait(lock);
     }
   }
-  if (stopped_) return DagBlock();
+  if (stopped_) return {std::make_shared<DagBlock>(), false};
 
-  auto blk = verified_qu_.begin()->second.front();
+  auto blk = std::make_shared<DagBlock>(verified_qu_.begin()->second.front());
```

Comment:
This calls DagBlock copy ctor thus making unnecessary copy. std::move could be used here if we add move semantics support to DagBlock class:
`  DagBlock(const DagBlock&) = default;
  DagBlock& operator=(const DagBlock&) = default;
  DagBlock(DagBlock&&) = default;
  DagBlock& operator=(DagBlock&&) = default;`